src/main/java/dao/H2Factory/H2UserDao.java

```diff
@@ -1,118 +1,71 @@
 package dao.H2Factory;
 
 import customerproductorder.models.User;
-import dao.DaoException;
-import dao.H2Factory.converters.UserConverter;
+import dao.H2Factory.utils.UserMapper;
 import dao.UserDaoInterface;
-import java.sql.Connection;
-import java.sql.PreparedStatement;
-import java.sql.ResultSet;
-import java.sql.SQLException;
-import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import java.util.Map;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Repository;
 
+@Repository("h2UserDao")
 class H2UserDao implements UserDaoInterface {
 
-    private final ConnectionProvider connectionProvider;
-    private static final Logger LOG = LoggerFactory.getLogger(H2UserDao.class);
-    private final UserConverter converter;
-
-    protected H2UserDao(ConnectionProvider connectionProvider) {
-        this.connectionProvider = connectionProvider;
-        converter = UserConverter.getInstance();
+    @Autowired
+    private NamedParameterJdbcTemplate jdbcTemplate;
+    H2UserDao() {
     }
 
-    public void create(User newUser) throws DaoException {
-        try (Connection connection = connectionProvider.getConnection()) {
-            PreparedStatement st = connection.prepareStatement("insert into "
-                    + "customerapplication.account(username,password) "
-                    + "values(?,?)");
-            st.setString(1, newUser.getUsername());
-            st.setString(2, newUser.getPassword());
-            st.executeUpdate();
-            connectionProvider.destroy();
-        } catch (SQLException ex) {
-            LOG.error(ex.getSQLState());
-            throw new DaoException("Inserting data error", ex);
-        }
+    public void create(User newUser) {
```

Comment:
It is good when create method either return id for created user or return created user entity.